Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -125,10 +125,19 @@ public class ConnectionPool {
         }
         item = pool[0]
         pool.removeFirst()
+        // Check if the item is alive, and append a new one to the pool if it isn't
+        if item.isConnected == false {
+            releaser(item)
+            if let newItem = generator() {
+                capacity += 1
```

Comment:
This still is not right.  We're removing a dead item from the pool, and because `releaser(item)` does not decrement the capacity, it is (temporarily) out of sync with the actual number items we expect to have.

We then expect to create a new one to replace it - key word being _replace_ - we do not need to increment the capacity.  However, this then yields the same logic flaw that we had before: if we simply remove the increment here, then if the generator fails, we're again out of sync.

I believe safer logic is this:
```swift
            releaser(item)
            capacity -= 1
            if let newItem = generator() {
                capacity += 1
```
If the generator succeeds, capacity is unchanged, we've swapped out a dead connection for a new one.  If it fails, capacity is reduced by 1.

Reply:
That makes sense, I've amended the logic to reflect that.